geoh5py/groups/property_group.py

```diff
@@ -68,7 +68,7 @@ def __init__(  # pylint: disable=too-many-arguments
         on_file: bool = False,
         uid: UUID | None = None,
         property_group_type: GroupTypeEnum | str = GroupTypeEnum.SIMPLE,
-        properties: list[UUID | Data | str] | None = None,
+        properties: Sequence[UUID | Data | str] | None = None,
```

Comment:
Changing to Sequence is better, but won't address the underlying issue of a mixed list

Reply:
With sequence, mypy does not complain for mix types.

There was not real underlying issue, just mypy complaining